src/seahawk_deck/seahawk_deck/motion_controller.py

```diff
@@ -18,43 +19,119 @@ def __init__(self):
-        """
+        """Called every time the joystick publishes a message. """
         self.get_logger().info(f"Joystick axes: {joy_msg.axes} buttons: {joy_msg.buttons}")
 
         # Compute desired motion in <x, y, z, r, p, y>
+
+        # CONTROLLER KEYMAPPINGS
+        controller = {
+            'axes': ['left_stick', 'right_stick'],
+            'left_stick': {
+                'x':        -joy_msg.axes[0],
+                'y':        joy_msg.axes[1],
+                'press':    joy_msg.buttons[9],
+            },
+            'right_stick': {
+                'x':        -joy_msg.axes[3],
+                'y':        joy_msg.axes[4],
+                'press':    joy_msg.buttons[10],
+            },
+            'left_trigger': joy_msg.axes[2],
+            'right_trigger':joy_msg.axes[5],
+            'dpad': {
+                'up':       max(joy_msg.axes[7], 0), # +
+                'down':     min(joy_msg.axes[7], 0), # -
+                'right':    max(joy_msg.axes[6], 0), # +
+                'left':     min(joy_msg.axes[6], 0), # -
+            },
+            'a':            joy_msg.buttons[0],
+            'b':            joy_msg.buttons[1],
+            'x':            joy_msg.buttons[2],
+            'y':            joy_msg.buttons[3],
+            'left_bumper':  joy_msg.buttons[4],
+            'right_bumper': joy_msg.buttons[5],
+            'window':       joy_msg.buttons[6],
+            'menu':         joy_msg.buttons[7],
+            'xbox':         joy_msg.buttons[8],
+        }
+
+        # BINDINGS
         twist_msg = Twist()
-        twist_msg.linear.x  = 0.0 # X 
-        twist_msg.linear.y  = 0.0 # Y 
-        twist_msg.linear.z  = 0.0 # Z 
-        twist_msg.angular.x = 0.0 # R 
-        twist_msg.angular.y = 0.0 # P 
-        twist_msg.angular.z = 0.0 # Y 
-
-        # Send the twist message for debugging.
+        twist_msg.linear.x  = controller['left_stick']['y'] # X (forwards)
+        twist_msg.linear.y  = controller['left_stick']['x']# Y (sideways)
+        twist_msg.linear.z  = (controller['left_trigger'] - controller['right_trigger']) / 2 # Z (depth)
+        twist_msg.angular.x = 0.0 # R (roll)
+        twist_msg.angular.y = controller['right_stick']['y'] # P (pitch) 
+        twist_msg.angular.z = controller['right_stick']['x'] # Y (yaw)
+
+        # CONSTANT AXIS SCALING
+        AXIS_SCALE = {
+            'linear':  {'x': 1, 'y': 1, 'z': 1},    # forwards, sideways, depth
+            'angular': {'x': 0, 'y': 0.5, 'z': 0.5},  # roll, pitch, yaw
+        }
+        twist_msg.linear.y  *= AXIS_SCALE['linear']['x']
+        twist_msg.linear.x  *= AXIS_SCALE['linear']['y']
+        twist_msg.linear.z  *= AXIS_SCALE['linear']['z']
+        twist_msg.angular.x *= AXIS_SCALE['angular']['x']
+        twist_msg.angular.y *= AXIS_SCALE['angular']['y']
+        twist_msg.angular.z *= AXIS_SCALE['angular']['z']
+
         self.twist_pub.publish(twist_msg)
```

Comment:
Does anything subscribe to the Twist messages? It looks like we're sending the throttle values along and the ROV will subscribe directly to them. If so, we can skip the pilot input message publishing in it's entirety.

Reply:
Twist messages are for debug and visualization.
Remember ros doesn't do anything with publishers unless someone is subscribed to them so having them doesn't hurt anything. Plus this is on the deck box

Reply:
I'm thinking it would be a good idea to leave the twist as it's own thing, as I said previously I did want to try and move the input code into its own file. Regardless of whether or not we do that, I don't think it would hurt anything to keep using it.

Reply:
Further debate and commentary on this can take place in #44